drivers/can/can_sam.c

```diff
@@ -0,0 +1,239 @@
+	.register_state_change_isr = can_sam_register_state_change_isr,
+	.timing_min = {
+		.sjw = 0x7f,
+		.prop_seg = 0x00,
+		.phase_seg1 = 0x01,
+		.phase_seg2 = 0x01,
+		.prescaler = 0x01
+	},
+	.timing_max = {
+		.sjw = 0x7f,
+		.prop_seg = 0x00,
+		.phase_seg1 = 0x100,
+		.phase_seg2 = 0x80,
+		.prescaler = 0x200
+	},
+#ifdef CONFIG_CAN_FD_MODE
+	.timing_min_data = {
+		.sjw = 0x01,
+		.prop_seg = 0x01,
+		.phase_seg1 = 0x01,
+		.phase_seg2 = 0x01,
+		.prescaler = 0x01
+	},
+	.timing_max_data = {
+		.sjw = 0x10,
+		.prop_seg = 0x00,
+		.phase_seg1 = 0x20,
+		.phase_seg2 = 0x10,
+		.prescaler = 0x20
+	}
+#endif
+};
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(can0), okay)
+
+static void config_can_0_irq();
+
+static const struct can_sam_config can_sam_cfg_0 = {
+	.pmc_id = DT_PROP(DT_NODELABEL(can0), peripheral_id),
+	/*.pin_list = {
+		ATMEL_SAM_DT_PIN(0, 0),
+		ATMEL_SAM_DT_PIN(0, 1)
+	},*/
+	.config_irq = config_can_0_irq,
+	.mcan_cfg = {
+		.can = (struct can_mcan_reg  *)
+			DT_REG_ADDR_BY_NAME(DT_NODELABEL(can0), m_can),
+		.bus_speed = DT_PROP(DT_NODELABEL(can0), bus_speed),
+		.sjw = DT_PROP(DT_NODELABEL(can0), sjw),
+#if DT_PROP(DT_NODELABEL(can0), sample_point) > 0
+		.sample_point = DT_PROP(DT_NODELABEL(can0), sample_point),
+#endif
+#if DT_PROP(DT_NODELABEL(can0), phase_seg1) > 0 && \
+    DT_PROP(DT_NODELABEL(can0), phase_seg2) > 0
+		.prop_ts1 = DT_PROP(DT_NODELABEL(can0), prop_seg) +
+			DT_PROP(DT_NODELABEL(can0), phase_seg1),
+		.ts2 = DT_PROP(DT_NODELABEL(can0), phase_seg2),
+#endif
+#ifdef CONFIG_CAN_FD_MODE
+		.bus_speed_data = DT_PROP(DT_NODELABEL(can0), bus_speed_data),
+		.sjw_data = DT_PROP(DT_NODELABEL(can0), sjw_data),
+#if DT_PROP(DT_NODELABEL(can0), sample_point_data) > 0
+		.sample_point_data =
+				DT_PROP(DT_NODELABEL(can0), sample_point_data),
+#endif
+#if DT_PROP(DT_NODELABEL(can0), phase_seg1_data) > 0 && \
+    DT_PROP(DT_NODELABEL(can0), phase_seg2_data) > 0
+		.prop_ts1_data = DT_PROP(DT_NODELABEL(can0), prop_seg_data) +
+				DT_PROP(DT_NODELABEL(can0), phase_seg1_data),
+		.ts2_data = DT_PROP(DT_NODELABEL(can0), phase_seg2_data),
+#endif
+#endif /* CONFIG_CAN_FD_MODE */
+	}
+};
+
+static struct can_sam_data can_sam_dev_data_0;
+
+DEVICE_AND_API_INIT(can_mcan_0, DT_LABEL(DT_NODELABEL(can0)), &can_sam_init,
+		    &can_sam_dev_data_0, &can_sam_cfg_0,
+		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
+		    &can_api_funcs);
+
+static void config_can_0_irq(void)
+{
+	LOG_DBG("Enable CAN0 IRQ");
+
+	IRQ_CONNECT(DT_IRQ_BY_NAME(DT_NODELABEL(can0), line_0, irq),
+		    DT_IRQ_BY_NAME(DT_NODELABEL(can0), line_0, priority),
+		    can_sam_line_0_isr, DEVICE_GET(can_mcan_0), 0);
+	irq_enable(DT_IRQ_BY_NAME(DT_NODELABEL(can0), line_0, irq));
+
+	IRQ_CONNECT(DT_IRQ_BY_NAME(DT_NODELABEL(can0), line_1, irq),
+		    DT_IRQ_BY_NAME(DT_NODELABEL(can0), line_1, priority),
+		    can_sam_line_1_isr, DEVICE_GET(can_mcan_0), 0);
+	irq_enable(DT_IRQ_BY_NAME(DT_NODELABEL(can0), line_1, irq));
+}
```

Comment:
This drive should be must instance because we have can0 and can1.
Try follow drivers/spi/spi_sam.c and add
`DT_INST_FOREACH_STATUS_OKAY(SPI_SAM_DEVICE_INIT)`